### DISCUSSION TITLE
Add support for emulating the original Xbox

### DISCUSSION THREAD

**Discussion by upintheairsheep** [PCBox/PCBox#93] [Ideas]
This might be a massive endeavour, but please consider adding support for original Xbox emulation, as it is based on the Pentium III and would likely be way better in terms of accuracy compared to xemu. See xemu for reverse engineering of the Xbox in QEMU.

**Comment by bachimusprime** [comment]
1) I was already planning on doing this, but it's currently blocked by my nVidia RIVA 128 and RIVA TNT emulation not being up to snuff yet.
2) It would also be FAR slower than xemu, to the point where you'd need AT LEAST a zen 5 CPU with 16 cores and 32 threads to use it at full speed.

**Reply by upintheairsheep** [reply]
> * I was already planning on doing this, but it's currently blocked by my nVidia RIVA 128 and RIVA TNT emulation not being up to snuff yet.
> * It would also be FAR slower than xemu, to the point where you'd need AT LEAST a zen 5 CPU with 16 cores and 32 threads to use it at full speed.

I know it would be very slow and pentium iii wasn’t added to regular 86box due to speed, and commercial CPUs being bad right now, but once it’s implemented and as CPUs improve, we’ll get to use this at full speed on a regular PC in the future.

**Reply by pankoza2-pl** [reply]
GeForce 3 emulation can be ported over from Bochs now, but even if PCBox implements the Original Xbox, the QEMU-based Xemu emulator will still be the preferred option for emulating the original Xbox

**Reply by bachimusprime** [reply]
I mean, think of it like xemu being like SNES9x and PCBox being more like bsnes. xemu is faster, but PCBox would be more accurate. :3 
Also, Bochs's Geforce 3 emulation is sadly only single-threaded, which would make it WAY too slow for the purposes of PCBox. I'll try to write something faster using multi-threading (Probably something able to use up to 16 threads, if I'm being honest), so stay tuned.
